run_all_tools.sh

```diff
@@ -1,4 +1,61 @@
+}
+
+# tfsec
+function run_tfsec {
+  echo Now running tfsec on all cases
+  docker pull tfsec/tfsec:latest
+  docker run -t -v $PWD:/tf tfsec/tfsec --version > version_tfsec.txt
+  find . -name "main.tf" -exec dirname {} \; | grep -v "\.terraform" | while read -r test_case; do echo $test_case ; ORG_PATH=$PWD ; cd $test_case ; if [ ! -f tfsec_results.txt ]; then docker run --rm -v "$(pwd):/src" tfsec/tfsec /src --no-color | sed "s~$ORG_PATH~tool-compare~" > tfsec_results.txt ; fi; cd $ORG_PATH; done
+}
+
+# KICS
+function run_kics {
+  echo Now running KICS on all cases
+  docker pull checkmarx/kics:latest
+  docker run -t -v $PWD:/tf checkmarx/kics version | awk '{print $NF}' > version_kics.txt
+  find . -name "main.tf" -exec dirname {} \; | grep -v "\.terraform" | while read -r test_case; do echo $test_case ; ORG_PATH=$PWD ; cd $test_case ; if [ ! -f kics_results.txt ]; then docker run --rm -v "$(pwd):/src" checkmarx/kics:latest -p /src | sed "s~$ORG_PATH~tool-compare~" | grep -v "Executing queries" > kics_results.txt ; fi ; cd $ORG_PATH; done
+}
+
+# Terrascan
+function run_terrascan {
+  echo Now running Terrascan on all cases
+  brew install terrascan
+  terrascan version | awk '{print $NF}' > version_terrascan.txt
+  find . -name "main.tf" -exec dirname {} \; | grep -v "\.terraform" | while read -r test_case; do echo $test_case ; ORG_PATH=$PWD ; cd $test_case ; if [ ! -f terrascan_results.txt ]; then terrascan scan | sed "s~$ORG_PATH~tool-compare~" > terrascan_results.txt ; fi ; cd $ORG_PATH; done
+}
+
+# Snyk
+function run_snyk {
+  echo Now running Snyk on all cases
+  docker pull snyk/snyk-cli:docker
+  docker run -t -v empty:/project -e SNYK_TOKEN snyk/snyk-cli:docker --version | tail -n 1 > version_snyk.txt
+  find . -name "main.tf" -exec dirname {} \; | grep -v "\.terraform" | while read -r test_case; do echo $test_case ; ORG_PATH=$PWD ; cd $test_case ; if [ ! -f snyk_results.txt ]; then docker run --rm -v "$(pwd):/project" -e SNYK_TOKEN snyk/snyk-cli:docker iac test --experimental /project | sed "s~$ORG_PATH~tool-compare~" > snyk_results.txt ; fi ; cd $ORG_PATH; done
+}
+
+# Cloudrail
+function run_cloudrail {
+  echo Now running Cloudrail on all cases
+  docker pull indeni/cloudrail-cli:latest
+  docker run -t -v $PWD:/tf indeni/cloudrail-cli --version | awk '{print $NF}' > version_cloudrail.txt
+  find . -name "main.tf" -exec dirname {} \; | grep -v "\.terraform" | while read -r test_case; do echo $test_case ; ORG_PATH=$PWD ; cd $test_case ; if [ ! -f cloudrail_results.txt ]; then docker run --rm -v $PWD:/data -e CLOUDRAIL_API_KEY indeni/cloudrail-cli run --tf-plan plan.out --output-file cloudrail_results.txt --no-cloud-account --auto-approve -v; fi ; cd $ORG_PATH; done
+}
+
+case $1 in
```

Comment:
This needs to be moved to the bottom. Some of the tools require the plan.out files that are generated further below.

Also, run_snyk requires the SNYK_TOKEN, and run_cloudrail requires the CLOUDRAIL_API_KEY. You could move those if's into their respective functions.

Reply:
Yup, that makes sense - didn't appreciate the relevance of the plan generation. Fixed